dataStorage/storage.js

```diff
@@ -67,17 +67,24 @@ async function loadFromCloudAsync(model, id) {
     try {
         // fetch data from service
         const response = await fetch(url);
-        // FIXME: [Wei] "response.json()" triggers error on Android, so I have to use "eval"
-        // responseJson = eval("(" + response._bodyText + ")")
-        responseString = await response.text();
-        responseJson = JSON.parse(responseString);
+
+        try {
+            // FIXME: [Wei] "response.json()" triggers error on Android, so I have to use "eval"
+            // responseJson = eval("(" + response._bodyText + ")")
+            responseString = await response.text();
+            responseJson = JSON.parse(responseString);
+        } catch (err) {
+            console.warn(err);
+            // FIXME: [Wei] "response.json()" triggers error on Android, so I have to use "eval"
+            // Fallback to eval workaround if JSON.parse() doesn't work
+            responseJson = eval("(" + responseString + ")");
+        }
 
         console.log(url + " => " + JSON.stringify(responseJson));
     } catch (err) {
-        console.warn(err);
-        // FIXME: [Wei] "response.json()" triggers error on Android, so I have to use "eval"
-        // Fallback to eval workaround if JSON.parse() doesn't work
-        responseJson = eval("(" + responseString + ")");
+        if (!silentLoad) {
+            alert("Failed to get data from network, check your network connection.");
```

Comment:
Add additional information from "err“ ?

alert("Failed to get data from network, check your network connection." + err);